### PR TITLE
Add pruning statistics docs

### PR DESCRIPTION
**Pull request by evgeniya-egupova**
### Changes

Added new section to documentation that describes filter pruning algorithm statistics

### Reason for changes

To provide a better understanding of algorithm logs.

### Related tickets

- #1011 
- 73888

### Tests

-

### CODE REVIEW

File: docs/compression_algorithms/Pruning.md [hunk 1]
```diff
@@ -27,6 +27,32 @@ Where ![L_j](https://latex.codecogs.com/png.latex?L_j) is j-th convolutional lay
 
 Then during pruning filters with smaller ![G(F_i)](https://latex.codecogs.com/png.latex?G%28F_i%29) importance function will be pruned first.
 
+#### Algorithm statistics
+A model compression can be measured by two main metrics: filter pruning rate and FLOPs pruning rate. While 
+filter pruning rate shows the ratio of removed filters to the total number of filters in the model, FLOPs pruning rate
```
Comment (AlexanderDokuchaev) [inline docs/compression_algorithms/Pruning.md:32]: `removed filters to the total number of filters in the model` - it's actual if all_weights parameter is True.
For default it ratio of removed filters to the total number of filters in each prunable convolution.

**Comment by alexsu52** [comment]
@evgeniya-egupova please rebase your PR

File: docs/compression_algorithms/Pruning.md (hunk 1 above)
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:32]: Please rename `pruning rate` to `pruning level` as it was suggested in #1030

**Comment by daniil-lyakhov** [comment]
Please don't forget to describe pruned filters statistics and the way pruning statistics are changing in cases of different  pruning algo configuration (Target filter pruning level  / Target FLOPs pruning level).

Besides it would be great if you would add compression statistics description for all other compression algorithms (quantization, sparsity)!

**Comment by daniil-lyakhov** [comment]
Jenkins please retry a build

**Comment by daniil-lyakhov** [comment]
Thanks for a brief description! Could you please add an example of pruning statistics in format they logged in sample? 
I think it will make easier to cite docs for a part of pruning statistics.
```
Statistics of the pruned model:
+---------+---------+---------+---------------+
|    #    |  Full   | Current | Pruning level |
+=========+=========+=========+===============+
| GFLOPS  | 1.044   | 0.307   | 0.706         |
+---------+---------+---------+---------------+
| MParams | 143.653 | 124.875 | 0.131         |
+---------+---------+---------+---------------+
| Filters | 5504    | 1608    | 0.708         |
+---------+---------+---------+---------------+
Prompt: statistic pruning level = 1 - statistic current / statistic full.
Statistics of the filter pruning algorithm:
+---------------------------------------+-------+
|           Statistic's name            | Value |
+=======================================+=======+
| Filter pruning level in current epoch | 0.800 |
+---------------------------------------+-------+
| Target filter pruning level           | 0.800 |
+---------------------------------------+-------+
```

File: docs/compression_algorithms/Pruning.md [hunk 2]
```diff
@@ -27,6 +27,46 @@ Where ![L_j](https://latex.codecogs.com/png.latex?L_j) is j-th convolutional lay
...
+filter pruning level shows the ratio of removed filters to the total number of filters in the model, FLOPs pruning level 
+indicates how the removed filters affect the number of floating point operations required to run a model. 
+
+During the algorithm execution several compression statistics are available:
+
+`Filter pruning level` - percentage of filters removed from the model. Calculated as: 
+
+![Filter pruning level](https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D%20%5Cinline%20%5Ctextit%7Bfilter%20pruning%20rate%20%7D=%201%20-%20(%5Ctextit%7Bcurrent%20number%20of%20filters%20in%20the%20model%20%7D/%5Ctextit%7B%20total%20number%20of%20filters%20in%20the%20model%7D))
+
+> **NOTE**: All other pruning levels are calculated in a similar way.
```
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:41]: Please consider to make description of statistic pruning level general as it done in pruning stats

File: docs/compression_algorithms/Pruning.md [hunk 3]
```diff
@@ -27,6 +27,74 @@ Where ![L_j](https://latex.codecogs.com/png.latex?L_j) is j-th convolutional lay
 
 Then during pruning filters with smaller ![G(F_i)](https://latex.codecogs.com/png.latex?G%28F_i%29) importance function will be pruned first.
 
+#### Algorithm statistics
+A model compression can be measured by two main metrics: filter pruning level and FLOPs pruning level. While 
+filter pruning level shows the ratio of removed filters to the total number of filters in the model, FLOPs pruning level 
+indicates how the removed filters affect the number of floating point operations required to run a model. 
+
+During the algorithm execution several compression statistics are available:
```
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:35]: What do you think about this?
```suggestion
During the algorithm execution several compression statistics are available, for example:
```

File: docs/compression_algorithms/Pruning.md [hunk 4]
```diff
@@ -27,6 +27,74 @@ Where ![L_j](https://latex.codecogs.com/png.latex?L_j) is j-th convolutional lay
...
++---------------------------------------+-------+
+|           Statistic's name            | Value |
++=======================================+=======+
+| Filter pruning level in current epoch | 0.500 |
++---------------------------------------+-------+
+| Target filter pruning level           | 0.800 |
++---------------------------------------+-------+
+```
+
+##### Model statistics
```
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:58]:
```suggestion
##### Model statistics
Columns `Full` and `Current` represents the value of correspondent statistics in a original model and in a compressed one in current state accordingly.
```

Comment (daniil-lyakhov) [reply docs/compression_algorithms/Pruning.md:58]: How about to describe columns first, and then describe all the rows?

File: docs/compression_algorithms/Pruning.md [hunk 5]
```diff
@@ -27,6 +27,74 @@ Where ![L_j](https://latex.codecogs.com/png.latex?L_j) is j-th convolutional lay
...
 Then during pruning filters with smaller ![G(F_i)](https://latex.codecogs.com/png.latex?G%28F_i%29) importance function will be pruned first.
 
+#### Algorithm statistics
+A model compression can be measured by two main metrics: filter pruning level and FLOPs pruning level. While 
+filter pruning level shows the ratio of removed filters to the total number of filters in the model, FLOPs pruning level 
+indicates how the removed filters affect the number of floating point operations required to run a model. 
+
+During the algorithm execution several compression statistics are available:
+```
+Statistics of the pruned model:
```
Comment (alexsu52) [inline docs/compression_algorithms/Pruning.md:37]: What about statistics by pruned layers?

File: docs/compression_algorithms/Pruning.md [hunk 6]
```diff
@@ -27,6 +27,74 @@ Where ![L_j](https://latex.codecogs.com/png.latex?L_j) is j-th convolutional lay
 
 Then during pruning filters with smaller ![G(F_i)](https://latex.codecogs.com/png.latex?G%28F_i%29) importance function will be pruned first.
 
+#### Algorithm statistics
+A model compression can be measured by two main metrics: filter pruning level and FLOPs pruning level. While
```
Comment (AlexKoff88) [inline docs/compression_algorithms/Pruning.md:31]: If you open this doc now it looks ugly. The narrative breaks after Geometric Median section. It should be fixed. I also suggest moving all the auxiliary information, such as statistics description, to the bottom of the document.

**Comment by evgeniya-egupova** [comment]
@AlexKoff88 @alexsu52 @daniil-lyakhov This PR is ready for review. Please resolve your comments if they were fulfilled.

**Review by vshampor — APPROVED** [review]

File: docs/compression_algorithms/Pruning.md [hunk 7]
```diff
@@ -118,3 +118,86 @@ This approach allows pruning the model taking into account layer-specific sensit
...
+Statistics by pruned layers:
++----------------------+------------------+--------------+---------------------+
+|     Layer's name     |  Weight's shape  | Mask's shape |   Filter pruning    |
+|                      |                  |              |        level        |
++======================+==================+==============+=====================+
+| ConvBlock[conv1]/NNC | [192, 32, 1, 1]  | [192]        | 0.500               |
+| FConv2d[conv]        |                  |              |                     |
++----------------------+------------------+--------------+---------------------+
+| ConvBlock[conv2]/NNC | [384, 64, 1, 1]  | [384]        | 0.500               |
+
```
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:138]: Looks like part of table is missed

File: docs/compression_algorithms/Pruning.md [hunk 8]
```diff
@@ -118,3 +118,86 @@ This approach allows pruning the model taking into account layer-specific sensit
...
+| Target filter pruning level           | 0.800 |
++---------------------------------------+-------+
+```
+##### Layer statistics
+`Statistics by pruned layers` section lists names of all layers that will be pruned, shapes of their weight tensors, 
+shapes of pruning masks applied to respective weights and percentage of zeros in those masks. 
+
+##### Model statistics
+The columns `Full` and `Current` represent the values of the corresponding statistics in the original model and compressed one in the current state, respectively. 
+
```
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:165]:
```suggestion
The `Pruning level` column lists a ratio between values of full and current statistics in the correspondent rows defined by the formula 

![Statistic pruning level](https://latex.codecogs.com/png.image?\dpi{110}\text{statistic&space;pruning&space;level}&space;=&space;1&space;-&space;\text{statistic&space;current}&space;/&space;\text{statistic&space;full})
```

File: docs/compression_algorithms/Pruning.md [hunk 9]
```diff
@@ -118,3 +118,86 @@ This approach allows pruning the model taking into account layer-specific sensit
...
+`Filter pruning level` - percentage of filters removed from the model. Calculated as: 
+
+![Filter pruning level](https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D%20%5Cinline%20%5Ctextit%7Bfilter%20pruning%20level%7D%20=%201%20-%20(%5Ctextit%7Bcurrent%20number%20of%20filters%20in%20the%20model%20%7D%20/%20%5Ctextit%7B%20total%20number%20of%20filters%20in%20the%20model%7D))
+
+> **NOTE**: All other pruning levels are calculated in a similar way.
```
Comment (daniil-lyakhov) [inline docs/compression_algorithms/Pruning.md:170]:
```suggestion
```

File: docs/compression_algorithms/Pruning.md (hunk 8 above)
Comment (daniil-lyakhov) [reply docs/compression_algorithms/Pruning.md:165]: \text{statistic pruning level} = 1 - \text{statistic current} / \text{statistic full}

Comment (evgeniya-egupova) [inline docs/compression_algorithms/Pruning.md:165]:
```suggestion
The `Pruning level` column lists a ratio between values of full and current statistics in the correspondent rows defined by the formula 
![Statistic pruning level](https://latex.codecogs.com/png.image?\dpi{110}\text{statistic pruning level} = 1 - \text{statistic current} / \text{statistic full})
```

File: docs/compression_algorithms/Pruning.md (hunk 7 above)
Comment (daniil-lyakhov) [reply docs/compression_algorithms/Pruning.md]: It was a bit unclear for me, I think closing line or indicator like `...` or `etc` is more clear than previous option. I think current version is clear, thanks!

**Review by daniil-lyakhov — APPROVED** [review]

**Comment by daniil-lyakhov** [comment]
Jenkins please retry a build

File: docs/compression_algorithms/Pruning.md (hunk 4 above)
Comment (evgeniya-egupova) [reply docs/compression_algorithms/Pruning.md]: Done

File: docs/compression_algorithms/Pruning.md (hunk 5 above)
Comment (evgeniya-egupova) [reply docs/compression_algorithms/Pruning.md]: Done

File: docs/compression_algorithms/Pruning.md (hunk 6 above)
Comment (evgeniya-egupova) [reply docs/compression_algorithms/Pruning.md]: Done

**Review by AlexKoff88 — APPROVED** [review]

**Comment by alexsu52** [comment]
Jenkins please retry a build

**Review by alexsu52 — APPROVED** [review]

**Comment by daniil-lyakhov** [comment]
test this please

**Comment by daniil-lyakhov** [comment]
rebased and moved to #1236

**Pull request closed without merging**